osquery/config/tests/config_tests.cpp

```diff
@@ -565,4 +591,73 @@ TEST_F(ConfigTests, test_config_refresh) {
   FLAGS_config_accelerated_refresh = refresh_acceleratred;
   rf.registry("config")->remove("test");
 }
+
+TEST_F(ConfigTests, test_config_backup) {
+  const auto config_enable_backup_saved = FLAGS_config_enable_backup;
+  FLAGS_config_enable_backup = true;
+
+  get().reset();
+  const std::map<std::string, std::string> expected_config = {{"a", "b"},
+                                                              {"c", "d"}};
+  get().backupConfig(expected_config);
+  const auto config = get().restoreConfigBackup();
+  EXPECT_TRUE(config);
+  EXPECT_EQ(*config, expected_config);
+
+  FLAGS_config_enable_backup = config_enable_backup_saved;
+}
+
+TEST_F(ConfigTests, test_config_backup_disabled) {
+  const auto config_enable_backup_saved = FLAGS_config_enable_backup;
+
+  get().reset();
+
+  FLAGS_config_enable_backup = false;
+  const auto config = get().restoreConfigBackup();
+  EXPECT_FALSE(config);
+
+  FLAGS_config_enable_backup = config_enable_backup_saved;
+}
+
+TEST_F(ConfigTests, test_config_backup_integrate) {
+  const auto config_enable_backup_saved = FLAGS_config_enable_backup;
```

Comment:
BTW this is the type of thing that my comment above regarding the check inside the implementation prevents from being needed.